src/test/java/org/jabref/logic/openoffice/oocsltext/CSLFormatUtilsTest.java

```diff
@@ -0,0 +1,721 @@
+                                "    1 Smith Bill, Jones Bob, Williams Jeff. Title of the test entry. <i>BibTeX Journal</i> 2016;<b>34</b>(3):45–67. Doi: 10.1001/bla.blubb.\n" +
+                                "  \n"
+                ),
+
+                // Numeric in parentheses - "(1)"
+                Arguments.of(
+                        STYLE_LIST.stream().filter(e -> "American Chemical Society".equals(e.getTitle())).findAny().orElse(null),
+                        "  \n" +
+                                "    (1) Smith, B.; Jones, B.; Williams, J. Title of the Test Entry. <i>BibTeX Journal</i> <b>2016</b>, <i>34</i> (3), 45–67. https://doi.org/10.1001/bla.blubb.\n" +
+                                "  \n"
+                ),
+
+                // Numeric with right parenthesis - "1)"
+                Arguments.of(
+                        STYLE_LIST.stream().filter(e -> "Chemical and Pharmaceutical Bulletin".equals(e.getTitle())).findAny().orElse(null),
+                        "  \n" +
+                                "    1) Smith B., Jones B., Williams J., <i>BibTeX Journal</i>, <b>34</b>, 45–67 (2016).\n" +
+                                "  \n"
+                ),
+
+                // Numeric in superscript - "<SUPERSCRIPT>1"
+                Arguments.of(
+                        STYLE_LIST.stream().filter(e -> "American Institute of Physics 4th edition".equals(e.getTitle())).findAny().orElse(null),
+                        "  <sup>1</sup> B. Smith, B. Jones, and J. Williams, “Title of the test entry,” BibTeX Journal <b>34</b>(3), 45–67 (2016).\n"
+                )
+        );
+    }
+
+    /**
+     * Test to check correct transformation of raw CSL citation with a single entry generated by citeproc-java methods into OO-ready text.
+     * <p>
+     * <b>Precondition:</b> This test assumes that {@link CitationStyleGenerator#generateInText(List, String, CitationStyleOutputFormat, BibDatabaseContext, BibEntryTypesManager) generateInText} works as expected.
+     * </p>
+     *
+     * @param style the CSL style to test transformation on.
+     */
+    @ParameterizedTest
+    @MethodSource("rawCitationWithSingleEntryProvider")
+    void ooHTMLTransformFromCitationWithSingleEntry(CitationStyle style, String expected) throws IOException {
+        Citation citation = CitationStyleGenerator.generateInText(List.of(testEntry), style.getSource(), CSLFormatUtils.OUTPUT_FORMAT, context, bibEntryTypesManager);
+        String inTextCitationText = citation.getText();
+        String actual = CSLFormatUtils.transformHTML(inTextCitationText);
+        OOText ooText = OOText.fromString(actual);
+        assertEquals(OOText.fromString(expected), ooText);
+    }
+
+    static Stream<Arguments> rawCitationWithSingleEntryProvider() {
+        return Stream.of(
+
+                Arguments.of(
+                        STYLE_LIST.stream().filter(e -> "American Psychological Association 7th edition".equals(e.getTitle())).findAny().orElse(null),
```

Comment:
I think, an exception is ok here -> `.get()` instead of `orElse(null)`

The ertror that nothing is found in the style list should be risen as early as possible.

Reply:
Changed. Did the same in `CitationStyleGeneratorTest`.